### DISCUSSION TITLE
Unable to use social logins when using factoryFunction to provide for SocialAuthServiceConfig

### DISCUSSION THREAD

**Discussion by sgtonion** [abacritt/angularx-social-login#508] [Q&A]
Hi there, I am trying to fetch the clientIds (Google/Facebook/etc) which is stored in my db and pass it into the provider but it seems that it is not working
<img width="399" alt="Screenshot 2022-05-30 at 4 52 25 PM" src="https://user-images.githubusercontent.com/54050487/170955469-81e00ff0-4f70-41e7-b784-6187b7af3fb7.png">

It is fetching it correctly
<img width="1059" alt="Screenshot 2022-05-30 at 4 53 36 PM" src="https://user-images.githubusercontent.com/54050487/170955940-90487a91-09d4-49ad-946e-3e1bbc704e42.png">

But when i try to use this.socialAuthService.signIn it is giving me this error:
<img width="1056" alt="Screenshot 2022-05-30 at 4 56 48 PM" src="https://user-images.githubusercontent.com/54050487/170956318-f8f64a99-3cac-487e-897a-930b7757207d.png">

The implementation of the factory function i also presume is correct:
<img width="842" alt="Screenshot 2022-05-30 at 4 58 19 PM" src="https://user-images.githubusercontent.com/54050487/170956585-cdba45d9-e0f0-4544-acfa-4bb219c8f5fb.png">

**Comment by Heatmanofurioso** [comment]
Hey. `@abacritt/angularx-social-login:1.2.1` has been deployed, can you test it with the new version if the lib, please?

**Reply by sgtonion** [reply]
Hi @Heatmanofurioso I tried to install the latest update but I am getting this error. Any thoughts?
**Edit: I upgraded to angular-13 but same issue persists
<img width="985" alt="Screenshot 2022-07-05 at 12 42 43 PM" src="https://user-images.githubusercontent.com/54050487/177251033-ac055471-0c9c-4043-93c1-8a8a854ece93.png">

**Reply by anton-nikitsiuk** [reply]
I have upgraded to Angular 14 to fix this

**Reply by sgtonion** [reply]
@anton-nikitsiuk thanks alot.
Yes i upgrade to angular 14 and it started to work.
@Heatmanofurioso it seems the problem still remains I still cannot 
seem to load the clientIds Dynamically

<img width="591" alt="Screenshot 2022-07-20 at 3 46 30 PM" src="https://user-images.githubusercontent.com/54050487/179926990-3a4f1d0f-a515-493e-8f88-9b94fe9c5142.png">

Configuration:
<img width="370" alt="Screenshot 2022-07-20 at 3 51 11 PM" src="https://user-images.githubusercontent.com/54050487/179927313-d7c43982-6c06-4122-982e-b14f9ff287c4.png">

<img width="530" alt="Screenshot 2022-07-20 at 3 51 05 PM" src="https://user-images.githubusercontent.com/54050487/179927337-0c0e18f4-39df-455c-835b-ce6b6b9e2ac1.png">

any help would be appreciated. Thank you
